src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/service/UserRoleService.kt

```diff
@@ -64,6 +65,45 @@ class UserRoleService(
       // otherwise they can assign all roles that can be assigned to any of their groups
     } else roleRepository.findByGroupAssignableRolesForUserId(userId).toSet()
 
+  @Transactional
+  suspend fun addRolesByUserId(
+    userId: UUID,
+    roleCodes: List<String>,
+  ) {
+    // already checked that user exists
+    userRepository.findById(userId)?.let { user: User ->
+      maintainUserCheck.ensureUserLoggedInUserRelationship(user.name)
```

Comment:
this check : throws 403 , This needs adding in ApiResponse

Reply:
Added to swagger